dhis-2/dhis-e2e-test/src/test/java/org/hisp/dhis/tracker/deduplication/PotentialDuplicatesTests.java

```diff
@@ -29,99 +29,90 @@
-import org.junit.jupiter.params.provider.ValueSource;
 
 /**
  * @author Gintare Vilkelyte <vilkelyte.gintare@gmail.com>
  */
-public class PotentialDuplicatesTests
-    extends PotentialDuplicatesApiTest
+public class PotentialDuplicatesTests extends PotentialDuplicatesApiTest
 {
     @BeforeEach
     public void beforeEach()
     {
         loginActions.loginAsAdmin();
     }
 
-    @ParameterizedTest
-    @ValueSource( strings = { "OPEN", "INVALID", "MERGED" } )
-    public void shouldFilterByStatus( String status )
+    @Test
+    public void shouldFilterByStatus()
     {
-        String teiA = createTei();
-        String teiB = createTei();
+        // OPEN
+        potentialDuplicatesActions.createAndValidatePotentialDuplicate( createTei(), createTei() );
 
-        potentialDuplicatesActions.createPotentialDuplicate( teiA, teiB, status ).validate().statusCode( 200 );
+        // INVALID
+        String duplicateId = potentialDuplicatesActions.createAndValidatePotentialDuplicate( createTei(), createTei() );
 
-        ApiResponse response = potentialDuplicatesActions.get( "", new QueryParamsBuilder().add( "status=" + status ) );
-        response
+        potentialDuplicatesActions.update( duplicateId + "?status=" + "INVALID", new JsonObjectBuilder().build() )
             .validate()
-            .body( "identifiableObjects", hasSize( greaterThanOrEqualTo( 1 ) ) )
-            .body( "identifiableObjects.status", everyItem( equalTo( status ) ) );
-    }
+            .statusCode( 200 );
 
-    @Test
-    public void shouldReturnAllStatuses()
-    {
-        Arrays.asList( "OPEN", "MERGED", "INVALID" ).forEach( status -> {
-            String teiA = createTei();
-            String teiB = createTei();
+        // MERGED
+        String potentialDuplicate = potentialDuplicatesActions.createAndValidatePotentialDuplicate( createTei(),
+            createTei() );
 
-            potentialDuplicatesActions.createPotentialDuplicate( teiA, teiB, status ).validate().statusCode( 200 );
-        } );
+        potentialDuplicatesActions.autoMergePotentialDuplicate( potentialDuplicate ).validate()
+            .statusCode( 200 );
 
         potentialDuplicatesActions.get( "", new QueryParamsBuilder().add( "status=ALL" ) )
             .validate()
-            .body( "identifiableObjects", hasSize( greaterThanOrEqualTo( 1 ) ) )
+            .body( "identifiableObjects", hasSize( greaterThanOrEqualTo( 3 ) ) )
             .body( "identifiableObjects.status",
                 allOf( hasItem( "OPEN" ), hasItem( "INVALID" ), hasItem( "MERGED" ) ) );
+
+        potentialDuplicatesActions.get( "", new QueryParamsBuilder().add( "status=" + "OPEN" ) )
+            .validate()
+            .body( "identifiableObjects", hasSize( greaterThanOrEqualTo( 1 ) ) )
+            .body( "identifiableObjects.status", everyItem( equalTo( "OPEN" ) ) );
+
+        potentialDuplicatesActions.get( "", new QueryParamsBuilder().add( "status=" + "INVALID" ) )
```

Comment:
I don't really see where `INVALID` potential duplicate was created? Only 1 `MERGED` should exist here since you only create one potential duplicate and change the status to `MERGED`.